metricbeat/module/rabbitmq/connection/data.go

```diff
@@ -57,30 +56,33 @@ var (
 	}
 )
 
-func eventsMapping(content []byte, r mb.ReporterV2) {
+func eventsMapping(content []byte, r mb.ReporterV2, m *MetricSet) error {
 	var connections []map[string]interface{}
 	err := json.Unmarshal(content, &connections)
 	if err != nil {
-		logp.Err("Error: %+v", err)
-		r.Error(err)
-		return
+		return errors.Wrap(err, "error in unmarshal")
 	}
 
-	var errors multierror.Errors
 	for _, node := range connections {
-		err := eventMapping(node, r)
+		evt, err := eventMapping(node)
 		if err != nil {
-			errors = append(errors, err)
+			m.Logger().Errorf("error in mapping: %s", err)
```

Comment:
why do you need to log this here if you are reporting it next?

Reply:
Reporting the error will not log it. So the error reported is for the UI / Kibana, the one logged for the operator.

Not sure if we should really log this on the error level.

Reply:
Really? I figured if an event was getting thrown out that would be an error.

Reply:
In the past we had the problem that when some fields were missing, the `schema.Apply` returned an error which caused error logs inside. It meant just the event was not complete but it was not dropped. But since then @jsoriano made a few improvements to the schema handling. Please double check on where the errors come from and if these to an actual error happened (not a missing field) +1 on logging it.

Reply:
Is there a way to do that? The only time `eventMapping` will return an error is if `schema.Apply` fails, and that returns a multi-error structure.

Reply:
Looking around the source, the error from `schema.Apply()` is handled...inconsistently.

```
 grep -rn "err := schema\.Apply" . | wc -l
      14
grep -rn "_ := schema\.Apply" . | wc -l
      16
```

Reply:
So, we can mark certain values in the schema as required, but then that raises the issue of exactly what _is_ required.

Reply:
But I suppose we can set `FailOnRequired` and at least that'll improve things?

Reply:
Just curious how did you pick `vhost`, `host` and `node` in this case?  maybe `name` should be required as well?